datafusion/functions-nested/src/extract.rs

```diff
@@ -330,7 +330,8 @@ pub(super) struct ArraySlice {
 impl ArraySlice {
     pub fn new() -> Self {
         Self {
-            signature: Signature::variadic_any(Volatility::Immutable),
+            // TODO: This signature should use the actual accepted types, not variadic_any.
```

Comment:
I haven't been able to figure out a way using the current `TypeSignature` to create a type signature that accepts (any list type, i64, i64). My only idea is to extend `ArrayFunctionSignature` with a variant like `ArrayAndElements(NonZeroUsize)`.

Reply:
Another approach, which probably warrants it's own issue/PR, is to add something similar to PostgreSQL psuedo-types (https://www.postgresql.org/docs/current/datatype-pseudo.html). Then I can make a signature of something like
```Rust
TypeSignature::OneOf(vec![
    TypeSignature::Exact(vec![AnyArray, Int64, Int64]),
    TypeSignature::Exact(vec![AnyArray, Int64, Int64, Int64]),
])
```

Reply:
>  My only idea is to extend ArrayFunctionSignature with a variant like ArrayAndElements(NonZeroUsize)

I think this is better

Reply:
Just pushed an update to do this.